io/net/bonding.py

```diff
@@ -470,8 +459,17 @@ def test_cleanup(self):
                     self.log.warn("unable to bring to original state in peer")
                 time.sleep(self.sleep_time)
         self.error_check()
+
+        try:
+            NetworkInterface(self.bond_name, self.localhost).restore_from_backup()
+        except Exception:
+            self.log.info("backup file not availbale, could not restore file.")
         for host_interface in self.host_interfaces:
             networkinterface = NetworkInterface(host_interface, self.localhost)
+            try:
+                networkinterface.bring_up()
+            except Exception:
+                self.fail("Interface is taking long time to link up")
             if networkinterface.set_mtu("1500") is not None:
                 self.cancel("Failed to set mtu in host")
             try:
```

Comment:
Can we run the above changes, with and with out a sysfs bonding configuration file..   ?

Reply:
yes, it works for with and without a sysfs bonding file.